cli/src/cli_util.rs

```diff
@@ -2600,3 +2602,37 @@ impl CliRunner {
         exit_code
     }
 }
+
+fn map_clap_cli_error(
+    cmd_err: CommandError,
+    ui: &Ui,
+    layered_configs: &LayeredConfigs,
+) -> CommandError {
+    if let CommandError::ClapCliError { ref err, hint: _ } = cmd_err {
+        if let (Some(ContextValue::String(ref arg)), Some(ContextValue::String(ref value))) = (
+            err.get(ContextKind::InvalidArg),
+            err.get(ContextKind::InvalidValue),
+        ) {
+            if arg.as_str() == "--template <TEMPLATE>" && value.is_empty() {
+                // Suppress the error, it's less important than the original error.
+                if let Ok(template_aliases) = load_template_aliases(ui, layered_configs) {
+                    return CommandError::ClapCliError {
+                        err: err.clone(),
+                        hint: Some(format_symbol_names_hint(&template_aliases)),
+                    };
+                }
+            }
+        }
+    }
+    cmd_err
+}
+
+fn format_symbol_names_hint(template_aliases: &TemplateAliasesMap) -> String {
+    let mut names = template_aliases.symbol_names().collect_vec();
+    names.sort_unstable();
+    let mut hint = String::from("The following symbol names are defined:\n");
+    for name in names {
+        hint.push_str(&format!("- {name}\n"));
+    }
```

Comment:
nit: could be `sorted_unstable().map(..).join("\n")` to remove the trailing `"\n"`.